### DISCUSSION TITLE
"An unexpected error has occurred"

### DISCUSSION THREAD

**Discussion by Gizmotronn** [vercel/next.js#25837] [Help] [answered]
Hi all, I've been working on a react application to embed parts of an Airtable database depending on which user is logged in. When running on my `localhost:3000` on OS X Big Sur, the application works:

![Screen Shot 2021-06-07 at 12 13 37 pm](https://user-images.githubusercontent.com/31812229/120958274-e0268580-c789-11eb-9242-39a90677ac67.png)
<img width="1440" alt="Screen Shot 2021-06-07 at 12 13 20 pm (2)" src="https://user-images.githubusercontent.com/31812229/120958287-e9175700-c789-11eb-9cad-a42d14263ab2.png">

The code is located in [this repository](https://github.com/acord-robotics/api-heroku)

<details><summary>I then added it to my Vercel account:</summary>

```bash
…/api-heroku on 🌱 main [✘!?] via 🤖 v16.0.0 took 13s 
❯ vercel
Vercel CLI 22.0.1
🔍  Inspect: https://vercel.com/gizmotronn/gemreactportal/92WLPiSz9hE2HRsQKDJWsHFEP9E9 [1s]
✅  Preview: https://gemreactportal-gizmotronn.vercel.app [copied to clipboard] [47s]
📝  To deploy to production (gemreactportal.vercel.app), run `vercel --prod`
```

</details>

The current deployment is "https://vercel.com/gizmotronn/gemreactportal/EMhB2sHQnn8LkVEeuQd1UeDMxDgT" and the domain is "http://gemreactportal.vercel.app".

However, when visiting both links I simply get the message "An unexpected error has occurred".

I'm not sure what to do - the app works fine locally but has issues on Vercel.

-L

**Comment by wwwdepot** [comment, marked as answer]
You forgot to add environment variables. https://vercel.com/docs/environment-variables

```
Cannot read property 'ApiKeyID' of undefined
```

**Reply by Gizmotronn** [reply]
After moving the variables from `.env.local` to the environment section (even doing a new project straight from Git), it resulted in an error during the Build phase of the deployment. Here's the log:

```

10:32:03.958
WARNING: You should not upload the `.next` directory.
10:32:03.960
Installing dependencies...
10:32:05.036
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
10:32:17.060
npm ERR! code EEXIST
10:32:17.060
npm ERR! path /vercel/path1/node_modules/.bin/next
10:32:17.066
npm ERR! Refusing to delete /vercel/path1/node_modules/.bin/next: is outside /vercel/path1/node_modules/next and not a link
10:32:17.066
npm ERR! File exists: /vercel/path1/node_modules/.bin/next
10:32:17.066
npm ERR! Remove the existing file and try again, or run npm
10:32:17.066
npm ERR! with --force to overwrite files recklessly.
10:32:17.094
10:32:17.095
npm ERR! A complete log of this run can be found in:
10:32:17.095
npm ERR!     /vercel/.npm/_logs/2021-06-08T02_32_17_086Z-debug.log
10:32:17.121
Error: Command "npm install" exited with 1
```

**Reply by Gizmotronn** [reply]
Actually, it worked!

![Screen Shot 2021-06-08 at 10 42 38 am](https://user-images.githubusercontent.com/31812229/121114450-4a9cfb80-c846-11eb-9264-7b94d94a245f.png)

Had to change the build command to `npm run dev`

Thanks for your help!
